datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategyFactory.java

```diff
@@ -12,69 +12,49 @@
 
-  public static final String CFG_MAX_PARTITION_PER_TASK = "maxPartitionsPerTask";
-  public static final String CFG_PARTITIONS_PER_TASK = "partitionsPerTask";
-  public static final String CFG_PARTITION_FULLNESS_THRESHOLD_PCT = "partitionFullnessThresholdPct";
-  public static final String CFG_ENABLE_ELASTIC_TASK_ASSIGNMENT = "enableElasticTaskAssignment";
-  public static final String CFG_ZK_ADDRESS = "zkAddress";
-  public static final String CFG_ZK_SESSION_TIMEOUT = "zkSessionTimeout";
-  public static final String CFG_ZK_CONNECTION_TIMEOUT = "zkConnectionTimeout";
-  public static final String CFG_CLUSTER_NAME = "cluster";
-
-  public static final boolean DEFAULT_ENABLE_ELASTIC_TASK_ASSIGNMENT = false;
-
   @Override
   public AssignmentStrategy createStrategy(Properties assignmentStrategyProperties) {
-    VerifiableProperties props = new VerifiableProperties(assignmentStrategyProperties);
-    int cfgMaxTasks = props.getInt(CFG_MAX_TASKS, 0);
-    int cfgImbalanceThreshold = props.getInt(CFG_IMBALANCE_THRESHOLD, 0);
-    int cfgMaxParitionsPerTask = props.getInt(CFG_MAX_PARTITION_PER_TASK, 0);
-    // Set to Optional.empty() if the value is 0
-    Optional<Integer> maxTasks = cfgMaxTasks > 0 ? Optional.of(cfgMaxTasks) : Optional.empty();
-    Optional<Integer> imbalanceThreshold = cfgImbalanceThreshold > 0 ? Optional.of(cfgImbalanceThreshold)
-        : Optional.empty();
-    Optional<Integer> maxPartitions = cfgMaxParitionsPerTask > 0 ? Optional.of(cfgMaxParitionsPerTask) :
-        Optional.empty();
-    boolean enableElasticTaskAssignment = props.getBoolean(CFG_ENABLE_ELASTIC_TASK_ASSIGNMENT,
-        DEFAULT_ENABLE_ELASTIC_TASK_ASSIGNMENT);
-    int cfgPartitionsPerTask = props.getInt(CFG_PARTITIONS_PER_TASK, 0);
-    int cfgPartitionFullnessThresholdPct = props.getIntInRange(CFG_PARTITION_FULLNESS_THRESHOLD_PCT, 0, 0, 100);
-    Optional<Integer> partitionsPerTask = cfgPartitionsPerTask > 0 ? Optional.of(cfgMaxParitionsPerTask) :
-        Optional.empty();
-    Optional<Integer> partitionFullnessThresholdPct = cfgPartitionFullnessThresholdPct > 0 ?
-        Optional.of(cfgPartitionFullnessThresholdPct) : Optional.empty();
-    String cluster = props.getString(CFG_CLUSTER_NAME, null);
+    PartitionAssignmentStrategyConfig config = new PartitionAssignmentStrategyConfig(assignmentStrategyProperties);
 
-    // Create the ZooKeeper Client
+    boolean enableElasticTaskAssignment = config.isElasticTaskAssignmentEnabled();
+    // Create the zookeeper client
     Optional<ZkClient> zkClient = Optional.empty();
-    String zkAddress = props.getString(CFG_ZK_ADDRESS, null);
-    if (enableElasticTaskAssignment && StringUtils.isBlank(zkAddress)) {
-      LOG.warn("Disabling elastic task assignment as zkAddress is not present or empty");
+    try {
+      zkClient = constructZooKeeperClient(enableElasticTaskAssignment, config.getZkAddress(),
+          config.getZkSessionTimeout(), config.getZkSessionTimeout());
```

Comment:
You are using `config.getZkSessionTimeout` instead of `config.getZkConnectionTimeout` for the 4th field.

Can you check if these kind of errors can be caught using tests. This reflects that our existing tests are missing some checks.